src/core/ext/filters/client_channel/lb_policy/xds/xds_load_balancer_api.cc

```diff
@@ -18,12 +18,63 @@
 
 #include <grpc/support/port_platform.h>
 
-#include "pb_decode.h"
-#include "pb_encode.h"
 #include "src/core/ext/filters/client_channel/lb_policy/xds/xds_load_balancer_api.h"
 
 #include <grpc/support/alloc.h>
 
+namespace grpc_core {
+
+namespace {
+
+constexpr char kEdsTypeUrl[] =
+    "type.googleapis.com/envoy.api.v2.ClusterLoadAssignment";
+constexpr char kEndpointRequired[] = "endpointRequired";
+upb_alloc* g_arena = nullptr;
+
+void InitArena() {
+  g_arena = upb_arena_new();
+}
+
+upb_alloc* arena() {
+  static gpr_once once = GPR_ONCE_INIT;
+  gpr_once_init(&once, InitArena);
+  return g_arena;
+}
+
+}
+
+XdsDiscoveryRequest* XdsRequestCreate(const char* service_name) {
```

Comment:
@haberman This PR is still in progress, but I want to make sure I'm using upb as expected here. Does this look correct?